custodia/httpd/server.py

```diff
@@ -433,36 +462,72 @@ def pipeline(self, config, request):
 
 
 class HTTPServer(object):
+    handler = HTTPRequestHandler
 
     def __init__(self, srvurl, config):
         url = urlparse(srvurl)
-        address = unquote(url.netloc)
+        serverclass, address = self._get_serverclass(url)
+        if sd is not None:
+            address = self._get_systemd_socket(address)
+        self.httpd = serverclass(address, self.handler, config)
+
+    def _get_serverclass(self, url):
         if url.scheme == 'http+unix':
             # Unix socket
+            address = None
+            address = unquote(url.netloc)
+            if not address:
+                raise ValueError('Empty address {}'.format(url))
+            logger.info('Serving on Unix socket %s', address)
             serverclass = ForkingUnixHTTPServer
-            if os.path.exists(address):
-                os.remove(address)
         elif url.scheme == 'http':
-            host, port = address.split(":")
+            host, port = url.netloc.split(":")
             address = (host, int(port))
+            logger.info('Serving on %s (HTTP)', url.netloc)
             serverclass = ForkingHTTPServer
         elif url.scheme == 'https':
-            host, port = address.split(":")
+            host, port = url.netloc.split(":")
             address = (host, int(port))
+            logger.info('Serving on %s (HTTPS)', url.netloc)
             serverclass = ForkingTLSServer
         else:
             raise ValueError('Unknown URL Scheme: %s' % url.scheme)
-
-        logger.debug('Serving on %s', address)
-
-        self.httpd = serverclass(address,
-                                 HTTPRequestHandler,
-                                 config)
+        return serverclass, address
+
+    def _get_systemd_socket(self, address):
+        fds = sd.listen_fds()
+        if not fds:
+            return address
+        elif len(fds) > 1:
+            raise ValueError('Too many listening sockets', fds)
+
+        if isinstance(address, tuple):
+            port = address[1]
+            # systemd uses IPv6
+            if not sd.is_socket_inet(fds[0], family=socket.AF_INET6,
+                                     type=socket.SOCK_STREAM,
+                                     listening=True, port=port):
```

Comment:
And in 2016 we still get reports that machines configured to do exclusively IPv4 break with this or that software ...